src/main/kotlin/de/amplimind/codingchallenge/utils/ValidationUtils.kt

```diff
@@ -18,4 +19,22 @@ object ValidationUtils {
             throw EmailFormatException("Email is not valid")
         }
     }
+
+    /**
+     * validates a supplied string to check if it is a valid password
+     *
+     * @param the string to be checked if it is a valid password
+     *
+     * @throws PasswordValidationException if the supplied string is not a valid password throws this exception
+     */
+    fun validatePassword(password: String) {
+        if (password.length < 8) {
+            throw PasswordValidationException("Password must be at least 8 characters long")
+        }
+
+        val illegalcharacters = Regex(pattern = "[!@#$%^&*()_+\\-=\\[\\]{};':\"\\\\|,.<>/?]")
```

Comment:
Speicher das in ner Konstante dann muss das Regex nur einmal kompiliert werden. So wirds grad bei jedem Funtionsaufruf kompiliert. Und warum heißen die IlleagalCharacters wenn wir mindestens eins haben wollen? + Kannste das auch für nummern machen?

Reply:
Das mit den Nummern aber mitm Frontend absprechen bitte